### ISSUE TITLE
Excludes/includes at the method level

### ISSUE THREAD

**Issue by victornoel** [siom79/japicmp#51]
Hi,

It would be great if it was possible to specify excludes (and includes) at the level of the method, something like:
```package.Class.method``` or maybe even something more specific including parameters types.

The motivation is that we use your (wonderful I must say) tool (the maven plugin in particular) to prevent changes to an API and so we want to always fail a build if there was such changes.
But of course, sometimes, there is exceptions and these exceptions are for specific method.
So we would like to skip checks for these :)

Thanks!

**Comment by siom79** [comment]
Hi,

today I have started a first version of this feature on the branch [feature-exclinclextended](https://github.com/siom79/japicmp/tree/feature-exclinclextended).

If you want, you can check out the branch and try it:
```
git clone https://github.com/siom79/japicmp.git
git checkout feature-exclinclextended
mvn install
```

Basically the two options `--exclude` and `--include` now take either packages, classes, methods or fields that are specified using the javadoc syntax:
* package.to.include
* package.Class
* package.Class#method(int,long)
* package.Class#field

Let me know what you think.

**Comment by victornoel** [comment]
Thanks, I will try it during the week :)

**Comment by victornoel** [comment]
Hi, so I did some tests, but I can't make it works…

I put the following exclude:
```
<exclude>javax.jbi.messaging.MessageExchange#getFlowAttributes();javax.jbi.messaging.MessageExchange#setFlowAttributes(org.ow2.petals.commons.log.FlowAttributes)</exclude>
```

Which corresponds to the following code (javax.jbi.messaging.MessageExchange is an interface):
```
public FlowAttributes getFlowAttributes();
public void setFlowAttributes(FlowAttributes flowAttributes);
```

Also, I think that in the maven plugin, it would be better if the excludes (and includes actually) could be specified like this:
```
<excludes>
<exclude>javax.jbi.messaging.MessageExchange#getFlowAttributes()</exclude>
<exclude>javax.jbi.messaging.MessageExchange#setFlowAttributes(org.ow2.petals.commons.log.FlowAttributes)</exclude>
</excludes>
```

Thanks!

**Comment by victornoel** [comment]
And I forgot to say, this is the maven output:
```
[INFO] /home/vnoel/.m2/repository/javax/activation/activation/1.1.1/activation-1.1.1.jar; scope: compile
[INFO] Comparing /home/vnoel/.m2/repository/org/ow2/petals/petals-jbi/1.3.2/petals-jbi-1.3.2.jar with /home/vnoel/Linagora/Petals/dev/petals.git/tools-commons/petals-jbi/target/petals-jbi-1.3.3-SNAPSHOT.jar:
***  MODIFIED CLASS: PUBLIC FINAL javax.jbi.messaging.ExchangeStatus  (not serializable)
	+++  NEW METHOD: PUBLIC(+) boolean equals(java.lang.Object)
***! MODIFIED INTERFACE: PUBLIC ABSTRACT javax.jbi.messaging.MessageExchange  (not serializable)
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) org.ow2.petals.commons.log.FlowAttributes getFlowAttributes()
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) void setFlowAttributes(org.ow2.petals.commons.log.FlowAttributes)
```

**Comment by victornoel** [comment]
Ha, surprisingly enough, now with this new commit, it works as desired!
I don't know why it didn't work before, maybe a problem in the parsing of the methods when using ";"…

So this seems good to me (for my need at least), should I close this issue?

**Comment by siom79** [comment]
Great to hear that it now works for you. I will close the issue once it is released.

**Comment by siom79** [comment]
Released with version [0.5.0](https://github.com/siom79/japicmp/releases/tag/japicmp-base-0.5.0).
